### PR TITLE
Add die shaped loading spinners

### PR DESCRIPTION
**Pull request by afbreilyn**
Currently, when one clicks on a die to generate a number, a square spins.  Now: a shape corresponding to the die's 2D profile spins instead! 

Testing: I'm using property based testing, which (while kind of lame) works here since the set of die is limited. 

Drawbacks: The human eye doesn't understand how size works, so the shapes are not _exactly_ the same size.

### CODE REVIEW

**Comment by aloverso** [comment]
re your comment in `Loader.tsx`, you could define your shapeMap using a record type, ie: `Record<Die, string>`

File: frontend/src/Loader.tsx [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+import { Die } from "./domain/Die";
+
+export interface LoaderProps {
+  dieType: Die;
+}
+
+export const Loader = ({ dieType }: LoaderProps): ReactElement => {
+  // Type aliases cannot be used as a parameter type, so no [key: Die]
+  // in other words....no dice! £pun
+  const shapeMap: { [key: string]: string } = {
```
Comment (aloverso) [inline frontend/src/Loader.tsx]: could use `Record<Die, string>` instead to keep the typing

**Review by aloverso — CHANGES_REQUESTED** [review]
v cool

File: frontend/src/Loader.tsx (hunk 1 above)
Comment (afbreilyn) [reply frontend/src/Loader.tsx]: Changed!

Comment (afbreilyn) [reply frontend/src/Loader.tsx]: An issue with this approach is that: 
Screen Shot 2020-12-06 at 4 20 58 PM doesn't throw any errors.... so enforcing the first thing to be a `Die` doesn't actually do what we want it do it 😞

File: frontend/src/domain/Die.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
-export const ALL_DICE = ["d4", "d6", "d8", "d10", "d12", "d20"];
+export const ALL_DICE = ["d4", "d6", "d8", "d10", "d12", "d20"] as const;
 
-export type Die = typeof ALL_DICE[number];
+type ElementType<T extends ReadonlyArray<unknown>> = T extends ReadonlyArray<infer ElementType>
```
Comment (aloverso) [inline frontend/src/domain/Die.ts:3]: wat

Comment (aloverso) [reply frontend/src/domain/Die.ts:3]: is this saying... like, "if it's in the array of type T, it's a valid ElementType, else it's not a valid ElementType"
?

File: frontend/src/domain/Die.ts [hunk 3]
```diff
@@ -1,3 +1,7 @@
-export const ALL_DICE = ["d4", "d6", "d8", "d10", "d12", "d20"];
+export const ALL_DICE = ["d4", "d6", "d8", "d10", "d12", "d20"] as const;
```
Comment (aloverso) [inline frontend/src/domain/Die.ts]: why this change though?

File: frontend/src/domain/Die.ts (hunk 2 above)
Comment (afbreilyn) [reply frontend/src/domain/Die.ts:3]: yeah! that's my understanding at least.

File: frontend/src/domain/Die.ts (hunk 3 above)
Comment (afbreilyn) [reply frontend/src/domain/Die.ts]: oops - it's gone now!

**Comment by aloverso** [comment]
LGTM fam
